src/App.tsx

```diff
@@ -1,24 +1,135 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 
-const USER_ID = 0;
+import { getTodos, deleteTodos } from './api/todos';
+import { Header } from './Components/Header';
+import { TodoList } from './Components/TodoList';
+import { Filter } from './Components/Filter';
+import { Error } from './Components/Error';
+import { Todo, TodoStatus } from './types/Todo';
+import { ErrorMessage } from './types/ErrorMessage';
+import { filterTodos } from './Components/filterTodo';
+import { UserIdContext } from './utils/context';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState(TodoStatus.All);
+  const [errorType, setErrorType] = useState(ErrorMessage.None);
+  const [isErrorShown, setIsErrorShown] = useState(false);
+  const [creatingTodoTitle, setCreatingTodoTitle] = useState('');
+  const [isClearCompleted, setIsClearCompleted] = useState(false);
+
+  const userId = useContext(UserIdContext);
+
+  useEffect(() => {
+    getTodos(userId)
+      .then((userTodos) => setTodos(userTodos))
+      .catch(() => {
+        setErrorType(ErrorMessage.Download);
+        setIsErrorShown(true);
+      });
+  }, []);
+
+  const counterActiveTodos = useMemo(
+    () => todos.reduce((num, todo) => {
+      return todo.completed ? num : num + 1;
+    }, 0),
+    [todos],
+  );
+
+  const counterCompletedTodos = todos.length - counterActiveTodos;
```

Comment:
u use these counters only for comparison with 0, why don't u use some just to make sure you have needed todos?